pkg/artifactory/repositories.go

```diff
@@ -1132,11 +1241,231 @@ func mkResourceSchema(skeema map[string]*schema.Schema, packer PackFunc, unpack
+		"composer":  "composer-default",
+		"conan":     "conan-default",
+		"conda":     "simple-default",
+		"debian":    "simple-default",
+		"docker":    "simple-default",
+		"gems":      "simple-default",
+		"generic":   "simple-default",
+		"gitlfs":    "simple-default",
+		"go":        "go-default",
+		"gradle":    "maven-2-default",
+		"helm":      "simple-default",
+		"ivy":       "ivy-default",
+		"maven":     "maven-2-default",
+		"npm":       "npm-default",
+		"nuget":     "nuget-default",
+		"opkg":      "simple-default",
+		"p2":        "simple-default",
+		"pub":       "simple-default",
+		"puppet":    "puppet-default",
+		"pypi":      "simple-default",
+		"sbt":       "sbt-default",
+		//"vagrant":   "simple-default", //currently not available
+		"vcs": "simple-default", //Type is only available in Remote Repository
+		"rpm": "simple-default",
+	}
+
+	if selectRandom {
+		repoLayout = selectRandomFromMapOfStrings(remoteDefaultRepoLayoutMap)
+	} else {
+		if value, ok := remoteDefaultRepoLayoutMap[repoType]; ok {
+			repoLayout = value
+		}
+	}
+	return repoLayout
+}
+
+//Return the default repo layout of Virtual Repository type
+func getDefaultVirtualRepoLayoutRef(repoType string, opts ...bool) string {
+
+	selectRandom := isSelectRandom(opts...)
+	const defaultRepoLayout = "simple-default"
+	repoLayout := defaultRepoLayout
+
+	virtualDefaultRepoLayoutMap := map[string]string{
+		"alpine": "simple-default",
+		"bower":  "bower-default",
+		"cran":   "simple-default",
+		//"cargo":     "simple-default", //currently not available
+		"chef": "simple-default",
+		//"cocoapods": "simple-default", //currently not available
+		"composer": "composer-default",
+		"conan":    "conan-default",
+		"conda":    "simple-default",
+		"debian":   "simple-default",
+		"docker":   "simple-default",
+		"gems":     "simple-default",
+		"generic":  "simple-default",
+		"gitlfs":   "simple-default",
+		"go":       "go-default",
+		"gradle":   "maven-2-default",
+		"helm":     "simple-default",
+		"ivy":      "ivy-default",
+		"maven":    "maven-2-default",
+		"npm":      "npm-default",
+		"nuget":    "nuget-default",
+		"opkg":     "simple-default",
+		"p2":       "simple-default",
+		"pub":      "simple-default",
+		"puppet":   "puppet-default",
+		"pypi":     "simple-default",
+		"sbt":      "sbt-default",
+		//"vagrant":   "simple-default", //currently not available
+		"rpm": "simple-default",
+	}
+
+	if selectRandom {
+		repoLayout = selectRandomFromMapOfStrings(virtualDefaultRepoLayoutMap)
+	} else {
+		if value, ok := virtualDefaultRepoLayoutMap[repoType]; ok {
+			repoLayout = value
+		}
+	}
+	return repoLayout
+}
+
+//Return the default repo layout of Federated Repository type
+func getDefaultFederatedRepoLayoutRef(repoType string, opts ...bool) string {
+
+	selectRandom := isSelectRandom(opts...)
+	const defaultRepoLayout = "simple-default"
+	repoLayout := defaultRepoLayout
+
+	federatedDefaultRepoLayoutMap := map[string]string{
```

Comment:
Are these maps even different? I don't see how. And, I don't see why they should be

Reply:
The map is different
```
+------------------------------------------+-------------------------------------------+--------------------------------------------+----------------------------------------------+
|             Local Repository             |             Remote Repository             |             Virtual Repository             |             Federated Repository             |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Local Repository Type | repo_layout_ref  | Remote Repository Type | repo_layout_ref  | Virtual Repository Type | repo_layout_ref  | Federated Repository Type | repo_layout_ref  |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Alpine                | simple-default   | Alpine                 | simple-default   | Alpine                  | simple-default   | Alpine                    | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Bower                 | bower-default    | Bower                  | bower-default    | Bower                   | bower-default    | Bower                     | bower-default    |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| CRAN                  | simple-default   | CRAN                   | simple-default   | CRAN                    | simple-default   | CRAN                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Cargo                 | simple-default   | Cargo                  | simple-default   |                         |                  | Cargo                     | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Chef                  | simple-default   | Chef                   | simple-default   | Chef                    | simple-default   | Chef                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| CocoaPods             | simple-default   | CocoaPods              | simple-default   |                         |                  | CocoaPods                 | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Composer              | composer-default | Composer               | composer-default | Composer                | composer-default | Composer                  | composer-default |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Conan                 | conan-default    | Conan                  | conan-default    | Conan                   | conan-default    | Conan                     | conan-default    |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Conda                 | simple-default   | Conda                  | simple-default   | Conda                   | simple-default   | Conda                     | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Debian                | simple-default   | Debian                 | simple-default   | Debian                  | simple-default   | Debian                    | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Docker                | simple-default   | Docker                 | simple-default   | Docker                  | simple-default   | Docker                    | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Gems                  | simple-default   | Gems                   | simple-default   | Gems                    | simple-default   | Gems                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Generic               | simple-default   | Generic                | simple-default   | Generic                 | simple-default   | Generic                   | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| GitLfs                | simple-default   | GitLfs                 | simple-default   | GitLfs                  | simple-default   | GitLfs                    | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Go                    | go-default       | Go                     | go-default       | Go                      | go-default       | Go                        | go-default       |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Gradle                | maven-2-default  | Gradle                 | maven-2-default  | Gradle                  | maven-2-default  | Gradle                    | maven-2-default  |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Helm                  | simple-default   | Helm                   | simple-default   | Helm                    | simple-default   | Helm                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Ivy                   | ivy-default      | Ivy                    | ivy-default      | Ivy                     | ivy-default      | Ivy                       | ivy-default      |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Maven                 | maven-2-default  | Maven                  | maven-2-default  | Maven                   | maven-2-default  | Maven                     | maven-2-default  |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Npm                   | npm-default      | Npm                    | npm-default      | Npm                     | npm-default      | Npm                       | npm-default      |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| NuGet                 | nuget-default    | NuGet                  | nuget-default    | NuGet                   | nuget-default    | NuGet                     | nuget-default    |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Opkg                  | simple-default   | Opkg                   | simple-default   | Opkg                    | simple-default   | Opkg                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
|                       |                  | P2                     | simple-default   | P2                      | simple-default   |                           |                  |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Pub                   | simple-default   | Pub                    | simple-default   | Pub                     | simple-default   | Pub                       | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Puppet                | puppet-default   | Puppet                 | puppet-default   | Puppet                  | puppet-default   | Puppet                    | puppet-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Pypi                  | simple-default   | Pypi                   | simple-default   | Pypi                    | simple-default   | Pypi                      | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| SBT                   | sbt-default      | SBT                    | sbt-default      | SBT                     | sbt-default      | SBT                       | sbt-default      |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Vagrant               | simple-default   |                        |                  |                         |                  | Vagrant                   | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
|                       |                  | VCS                    | simple-default   |                         |                  |                           |                  |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
| Rpm                   | simple-default   | Rpm                    | simple-default   | Rpm                     | simple-default   | Rpm                       | simple-default   |
+-----------------------+------------------+------------------------+------------------+-------------------------+------------------+---------------------------+------------------+
```